pkg/distributor/ha_tracker.go

```diff
@@ -51,6 +52,78 @@ func NewReplicaDesc() *ReplicaDesc {
 	return &ReplicaDesc{}
 }
 
+// Merge merges other replica into this one. Returns a new replica that represents the change,
+// and can be sent out to other clients.
+func (r *ReplicaDesc) Merge(other memberlist.Mergeable, _ bool) (memberlist.Mergeable, error) {
```

Comment:
FYI: Merge function is not aligning with the requirements of the interface, needs to re-iterate. As a first step I just wanted to just enable ha_tracker to use memberlist.